### PR TITLE
MAINT: enforce torch version

### LINKED ISSUE #202: Check PyTorch version on import
We should check that the correct PyTorch version is installed and raise an error if it isn't.
See #201.

### PR DESCRIPTION
**Pull request by stsievert**
Closes #202 

This does require maintaining the minimum required torch version. I coded this as a `MIN_TORCH_VERSION` in `skorch/__init__.py`. Will that work?

I tested this with torch 0.3.1 and it threw the error. With torch 0.4.0 it didn't.

### CODE REVIEW

File: skorch/__init__.py [hunk 1]
```diff
@@ -1,6 +1,22 @@
...
+
+try:
+    import torch
+except:  # pylint: disable=bare-except
+    raise Exception('skorch depends on PyTorch. Visit https://pytorch.org/ '
+                    'for install instructions')
+
+torch_version = pkg_resources.get_distribution('torch').version
+if parse_version(torch_version) < parse_version(MIN_TORCH_VERSION):
+    msg = ('skorch depends a newer version of PyTorch (at least {req}, not '
```
Comment (githubnemo) [inline skorch/__init__.py]: depends *on*

File: skorch/__init__.py [hunk 2]
```diff
@@ -1,6 +1,22 @@
...
 import pkg_resources
+from pkg_resources import parse_version
+
+MIN_TORCH_VERSION = '0.4.0'
+
+try:
+    import torch
+except:  # pylint: disable=bare-except
+    raise Exception('skorch depends on PyTorch. Visit https://pytorch.org/ '
+                    'for install instructions')
```
Comment (githubnemo) [inline skorch/__init__.py]: install*ation*

File: skorch/__init__.py [hunk 3]
```diff
@@ -1,6 +1,22 @@
 """skorch base imports"""
 
 import pkg_resources
+from pkg_resources import parse_version
+
+MIN_TORCH_VERSION = '0.4.0'
+
+try:
+    import torch
+except:  # pylint: disable=bare-except
```
Comment (BenjaminBossan) [inline skorch/__init__.py]: I believe `except ModuleNotFoundError` would be enough here.

**Comment by BenjaminBossan** [comment]
Apart from comments, this looks good. `__init__.py` is the right place.

**Comment by stsievert** [comment]
Updated. I edited the error messages and exceptions raised to be more similar to Python's default.

**Comment by ottonemo** [comment]
LGTM. Thanks for contributing!

**Pull request merged**